### ISSUE TITLE
Proposal: user-friendly DPI setting.

### ISSUE THREAD

**Issue by ma-laforge** [JuliaPlots/Plots.jl#742]
Currently, Plots.jl specifies (mostly) all of its dimenensions in pixels.  For example, line widths & plot size are specified in pixels.

This solution is perfect for rending images on screen, because the user can really visualize how things will end up looking on their screen.

The problem is when people want to generate high-resolution "bitmap" images (typically .png).

### Solution:
 1. Assume all dimensions are in typography "points" (1/72")- instead of pixels.
 2. Anything that gets rendered to screen somehow gets rendered using a "dpi" value of 72 - irrespective of the DPI value assigned to the plot.  This way, users gets what they expect on screen - no surprises.  A width of 1 shows up as 1 pixel wide on the screen!
 3. When the user saves to a bitmap format, that's when they actually get to apply the dpi value in a way that is useful:
```
png("MyHighResPicture.png", dpi=4*72) #4*4=16x more pixels than on screen!
```

I'm not sure how to structure Plots.jl to do this...  But I think this solution would be very intuitive for most people - because it would actually do what they expect!

As a side note, the `dpi` value would not affect the output to vector graphics (like .svg) because those would get rendered exactly the same as what you see on screen... only with an "infinite" resolution!

**Comment by ma-laforge** [comment]
### Implication
This solution proposed above implies that we are now using a "virtual DPI" value:

 - We never really intended the plot sizes to have any meaning in the real world (i.e. we are not telling Plots.jl that we want to generate a plot that will be 8"x6").
 - Instead, we are starting off with a sufficiently fine unit of measurement (the "point"), and tying that to the width of a pixel.
 - Thus, when we then scale up/down the dpi value from 72 dpi - we are not working down from an intended physical dimension to figure out how many pixels to use.  Instead, we are working outwards from a pixel that happens to measure 1 "point".

That being said, I believe this "virtual DPI" method is much closer to what most users want/expect.

**Comment by spaceLem** [comment]
It's handy to be able to specify the dpi or size even when outputting as an svg, because you might need to know the size of the final figure and need a specific font size (probably 12pt).

**Comment by ma-laforge** [comment]
> @spaceLem: [...] you might need to know the size of the final figure and need a specific font size (probably 12pt).

Agreed with this part of your statement.

What I am saying in this thread is that we should use the "point" as the base unit for all dimensions in the plot image:
  - The point is a physical dimension that measures 1/72".
  - Since the font dimensions (like all dimensions) are in points, we can indeed specify a font size of 12pt.
  - Moreover, if we want our plot to measure 4"x6", we would set the plot `(w,h) = (4*72, 6*72)`.

At this point, DPI has no meaning to an SVG file.  SVGs simply describe how one would draw an image with objects (circles, arcs, lines, ...) of user-defined lengths, etc.

***Intrinsically, SVG has no concept of "a pixel" - so DPI is not relevant.***

The DPI value only comes into play when we render an image on a pixel-based canvas (like a PNG file, for example).  It is at that rendering step that we need to specify a DPI value.

**Comment by ma-laforge** [comment]
@spaceLem: I noticed my previous comments might have been a bit confusing, so I revised the second entry.

**Comment by spaceLem** [comment]
@ma-laforge Thanks. Mixing DPI and SVG was me being overly loose with my terms, I really just meant it's handy to be able to tie font sizes and line weights to the image dimensions, so you don't need to manually resize everything afterwards in Inkscape.

The SVG files I was making with Plots didn't seem to like me fiddling with any of the text, and it was easier to just resize the image, delete all the text and create new text boxes... which I suspect was unnecessary (unless the text was being saved as paths or something?)

**Comment by mkborregaard** [comment]
I agree with @spaceLem that the dpi could be equally useful in vector-based contexts in the same way that the `size` argument is used with vector formats - it sets the relationship between the image size and the size of image elements.

**Comment by ma-laforge** [comment]
Thanks for clarifying that @mkborregaard.  I thought @spaceLem was simply indicating that the plot backend being used did not respond to `font.pointsize` property.

So, in response to that, I opened a new issue: https://github.com/JuliaPlots/Plots.jl/issues/891.
